monai/losses/sure_loss.py

```diff
@@ -0,0 +1,214 @@
+        """
+        super().__init__()
+        self.perturb_noise = perturb_noise
+        self.eps = eps
+
+    def forward(
+        self,
+        operator: Callable,
+        x: torch.Tensor,
+        y_pseudo_gt: torch.Tensor,
+        y_ref: Optional[torch.Tensor] = None,
+        complex_input: Optional[bool] = False,
+    ) -> torch.Tensor:
+        """
+        Args:
+            operator (function): The operator function that takes in an input
+            tensor x and returns an output tensor y. We will use this to compute
+            the divergence. More specifically, we will perturb the input x by a
+            small amount and compute the divergence between the perturbed output
+            and the reference output
+
+            x (torch.Tensor): The input tensor of shape (B, C, H, W) to the
+            operator. C=1 or 2: For complex input, the shape is (B, 2, H, W) aka
+            C=2 real.  For real input, the shape is (B, 1, H, W) real.
+
+            y_pseudo_gt (torch.Tensor): The pseudo ground truth tensor of shape
+            (B, C, H, W) used to compute the L2 loss. C=1 or 2: For complex
+            input, the shape is (B, 2, H, W) aka C=2 real.  For real input, the
+            shape is (B, 1, H, W) real.
+
+            y_ref (torch.Tensor, optional): The reference output tensor of the
+            same shape as y_pseudo_gt
+
+        Returns:
+            sure_loss (torch.Tensor): The SURE loss scalar.
+        """
+
+        # check inputs
+        # dim check:
+        assert x.dim() == 4, "Input tensor x should be 4D."
+        assert y_pseudo_gt.dim() == 4, "Input tensor y_pseudo_gt should be 4D."
+        if y_ref is not None:
+            assert y_ref.dim() == 4, "Input tensor y_ref should be 4D."
+
+        # complex/real check:
+        if complex_input:
+            assert (
+                x.shape[1] == 2 and y_pseudo_gt.shape[1] == 2 and not x.is_complex() and not y_pseudo_gt.is_complex()
+            ), "For complex input, the shape is (B, 2, H, W) aka C=2 real or (B, 1, H, W) aka C=1 complex"
+            if y_ref is not None:
+                assert (
+                    y_ref.shape[1] == 2 and not y_ref.is_complex()
+                ), "For complex input, the shape is (B, 2, H, W) aka C=2 real or (B, 1, H, W) aka C=1 complex"
+        else:  # real input
+            assert (
+                x.shape[1] == 1 and y_pseudo_gt.shape[1] == 1 and not x.is_complex() and not y_pseudo_gt.is_complex()
+            ), "For real input, the shape is (B, 1, H, W) real."
+            if y_ref is not None:
+                assert y_ref.shape[1] == 1 and not y_ref.is_complex(), "For real input, the shape is (B, 1, H, W) real."
+
+        # shape check
+        assert x.shape == y_pseudo_gt.shape, "Input tensor x and y_pseudo_gt should have the same shape."
+        if y_ref is not None:
+            assert y_pseudo_gt.shape == y_ref.shape, "Input tensor y_pseudo_gt and y_ref should have the same shape."
```

Comment:
We haven't been using asserts any more in MONAI, I think these would be best as checks which raise `ValueError` or some other exception if the check fails.